src/components/AuthContextProvider.js

```diff
@@ -30,6 +31,21 @@ const AuthContextProvider = ({ children }) => {
       setErrorVisible(true);
     }
   };
+
+  const handleUserRegister = async (name, email, password) => {
+    try {
+      await userRegister(name, email, password);
```

Comment:
Register call should return jwt and user data, so no need to call register and then login :)